File: src/main/java/com/devspace/conexfy/converters/ConStringToConAuthTypeEnumReadingConverter.java
package com.devspace.conexfy.converters;

import org.springframework.core.convert.converter.Converter;
import org.springframework.data.convert.ReadingConverter;
import org.springframework.lang.NonNull;

import com.devspace.conexfy.enums.ConAuthTypeEnum;

@ReadingConverter
public class ConStringToConAuthTypeEnumReadingConverter implements Converter<String, ConAuthTypeEnum> {
    @Override
    public ConAuthTypeEnum convert(@NonNull String source) {
        return ConAuthTypeEnum.valueOf(source);
    }
}

File: src/main/kotlin/com/devspace/conexfy/converters/ConStringToConAuthTypeEnumReadingConverter.kt
package com.devspace.conexfy.converters

import com.devspace.conexfy.enums.ConAuthTypeEnum
import org.springframework.core.convert.converter.Converter
import org.springframework.data.convert.ReadingConverter
import org.springframework.lang.NonNull

@ReadingConverter
class ConStringToConAuthTypeEnumReadingConverter : Converter<String, ConAuthTypeEnum> {
    override fun convert(@NonNull source: String): ConAuthTypeEnum {
        return ConAuthTypeEnum.valueOf(source)
    }
}
